src/lib/server-api/server-connection.ts

```diff
@@ -51,50 +57,70 @@ export class ServerConnection {
     /**
      * Post a msg object
      */
-    public post<T extends Typehinted>(msg: any): PromiseLike<T> {
-        const p = Promise.defer<T>()
+    public async post<T extends Typehinted>(msg: any): Promise<T> {
+        let dResolve: any
+        let dReject: any
+        const p = new Promise<any>((resolve, reject) => {
+            dResolve = resolve
+            dReject = reject
+        })
+
+        const d: Deferred<T> = {
+            promise: p,
+            resolve: dResolve,
+            reject: dReject
+        }
+
```

Comment:
Yes, this is a bit of a hack I've got going on here. I'll have to stare at this some more to further before I answer your question.

Reply:
What is the harm in resolving `post` before ensuring the websocket client received the message?

Reply:
The way the message receive handler is defined is kind of a funky. I've been playing around with a modified ServerClient that owns the message receive handler and it's....better...but still kind of funky. It would be smoother to just resolve the post() after sending the client message. But I may be missing some domain knowledge as to why its not resolving until the ws has received the rpc response.

Reply:
> What is the harm in resolving `post` before ensuring the websocket client received the message?

Not sure I understand? The post is async and I thought it a good idea to use promises instead of callbacks in the client api so the post just hides the fact that sending a question and receiving the answer is completely decoupled and just matches on that id stuck as key in the callback map…

So what I'm saying now is that you can probably just stick the resolve-function instead of the deferred thingies and just call it with the json…?